src/test/scala/ExampleWorkflowSpec.scala

```diff
@@ -54,9 +54,11 @@ class ExampleWorkflowSpec extends AnyWordSpec with Matchers with BeforeAndAfterA
           "SHACL_DIRECTORY" -> shaclDirectoryPath,
           "SHACL_FILENAME"  -> shaclFilename
         )
-      val is = this.getClass.getClassLoader.getResourceAsStream(exampleWorkflow)
-      val _ = WorkflowManager.runTransformation(is, "src/test/resources", Some(params), Some(plugins))
-      is.close()
+      val workflowFile = Paths.get(this.getClass.getClassLoader.getResource(exampleWorkflow).toURI).toFile
+      val workflowParentDir = workflowFile.getParent
+      val workflowInputStream = new FileInputStream(workflowFile)
+      val _ = WorkflowManager.runTransformation(workflowInputStream, workflowParentDir, Some(params), Some(plugins))
+      workflowInputStream.close()
```

Comment:
61 and 59 need to be done in a way that the resource is always closed. May I suggest writing a utility function to wrap the management of that, e.g.:
```scala
@throws(classOf[java.io.IOException])
def manage[T <: java.io.Closeable, U](open: => T)(use: T => U) : U {
   val resource = open()
   try {
     use(resource)
   } finally {
     resource.close()
   }
}
```

You can then use it like:
```scala
val _ = manage(new FileInputStream(workflowFile)) { workflowInputStream =>
  WorkflowManager.runTransformation(workflowInputStream, workflowParentDir, Some(params), Some(plugins))
}
```

**NOTE:** the code above is written "off the top of my head", so the syntax may need tweaking...

Reply:
Ah nice! About time they had that in the standard library. Also cool to see that Using.apply has almost exactly the same signature I have been using for my `manage` function over the years